### PR TITLE
Reenable tests in Geometry2d::Rect

### LINKED ISSUE #1571: [TODO] Reenable tests in Geometry2d::Rect
The following should just be a FRIEND_TEST or something:
See that these tests have been commented out.
https://github.com/RoboJackets/robocup-software/blob/ca8ce05708875fa106109bfadd752f9f73c4cab3/Geometry2d/testing/RectTest.cpp#L38-L60

### PR DESCRIPTION
**Pull request by kasohrab**
## Description
Reenable tests in Geometry2d::Rect

## Associated Issue
Issue #1571 

## Steps to test

Expected result:

### CODE REVIEW

**Comment by kylestach** [comment]
@kasohrab if you take a look at the error list:
https://app.circleci.com/pipelines/github/RoboJackets/robocup-software/677/workflows/d948a21b-812f-4da2-8936-3c6a75736a3b/jobs/15406
you'll see that we're seeing an error because `Rect::CohenSutherlandOutCode` is private (remember that this means that it can't be accessed from things outside of the class - including tests).

We can let things (functions and other classes) have access to private variables by declaring them as `friend` (this wasn't in training, it's a pretty niche feature of the language). Specifically for tests, we can use the `FRIEND_TEST` macro (see documentation).

In this case, you'll need to add the following declarations to the `Rect` class (in the header):
```cpp
FRIEND_TEST(Rect, cohenCodes);
FRIEND_TEST(Rect, degenerageCohenCodes);
```

**Comment by kylestach** [comment]
My bad, I forgot to mention that `rect.hpp` needs to include `<gtest/gtest.h>`.

Tip to avoid waiting for long CI times: make sure things build and run on your VM before submitting!

**Review by kylestach — APPROVED** [review]
LGTM!

File: rj_geometry/testing/rect_test.cpp [hunk 1]
```diff
@@ -33,31 +33,29 @@ TEST(Rect, contains_point) {
...
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-.5,0)), 0x04);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-1,0)), 0x04);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-1,3)), 0x08);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-.5,3)), 0x08);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-1.1,0)), 0x05);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(1.1,0)), 0x06);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(-1.5,2.5)), 0x09);
-    EXPECT_EQ(intersect_example.CohenSutherlandOutCode(Point(1.5,2.5)), 0x0A);
-}*/
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(0,1.2)), 0x00);
```
Comment (kylestach) [inline rj_geometry/testing/rect_test.cpp]: This should be back to `intersect_example`

File: doc/PlaysTacticsSkillsSummary.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ For higher level plays, only Coordinated Pass and Defense are really used.
 
 | Tactics                | Description | Use When ... |
 | ---------------------- | ----------- | ------------ |
-| Advance Zone Midfielder| Moves one robot to the best position to recieve the ball, and moves another robot to the best position to recieve the ball from that position. | Use it when need to move robots on a downfield attack at goal. |
+| Advance Zone Midfielder| Moves one robot to the best position to receive the ball, and moves another robot to the best position to receive the ball from that position. | Use it when need to move robots on a downfield attack at goal. |
 | Coordinated Pass       | Moves one robot to the ball and the other to the receive point. When they both are ready, they will pass | Use when you want to pass the ball from one robot to another |
 | Defense                | Basic 2 robot defense that stays on the goalie box. Clears the ball when it comes close. | Use in most cases unless you are writing your own defense |
 | Defense Old            | Same as Defense, but significantly harder to read code-wise | Don't use |
```
Comment (spatel776) [inline doc/PlaysTacticsSkillsSummary.md:21]: Same as **Defense**, but significantly harder to read code-wise

File: doc/PlaysTacticsSkillsSummary.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ For higher level plays, only Coordinated Pass and Defense are really used.
 
 | Tactics                | Description | Use When ... |
 | ---------------------- | ----------- | ------------ |
-| Advance Zone Midfielder| Moves one robot to the best position to recieve the ball, and moves another robot to the best position to recieve the ball from that position. | Use it when need to move robots on a downfield attack at goal. |
+| Advance Zone Midfielder| Moves one robot to the best position to receive the ball, and moves another robot to the best position to receive the ball from that position. | Use it when need to move robots on a downfield attack at goal. |
 | Coordinated Pass       | Moves one robot to the ball and the other to the receive point. When they both are ready, they will pass | Use when you want to pass the ball from one robot to another |
 | Defense                | Basic 2 robot defense that stays on the goalie box. Clears the ball when it comes close. | Use in most cases unless you are writing your own defense |
```
Comment (spatel776) [inline doc/PlaysTacticsSkillsSummary.md:20]: **Basic 2** robot defense that stays on the goalie box. Clears the ball when it comes close

File: doc/UnitTestsAndCI.md [hunk 4]
```diff
@@ -18,10 +18,10 @@ We use the standard `unittest` module for testing our python code.  Our unit tes
 
 ## Continuous Integration
 
-We use a free continuous integration service called [CircleCi](http://circleci.com), which recompiles and retests our project everytime we push a new commit to our [GitHub repo](http://github.com/robojackets/robocup-software).  Circle-ci works by cloning our git repository into a Ubuntu [Docker](https://www.docker.com) Container (similar to a Virtual Machine) each time it receives a notification from GitHub that there are new commits.  It then looks at our circle config file [circle.yml](../circle.yml), which runs other scripts that build and test our code.  Robocup-Software uses [DoCIF](https://github.com/jgkamat/DoCIF) to manage it's docker-based CI.  Triggered by the circle.yml file, DoCIF builds a baseimage for the project (if any setup files changed), runs the tests listed in [config.docif](../config.docif), and sends it's results back to github to be displayed.  On pull requests, you can see the individual logs of any test by clicking 'details' on any check.
```
Comment (spatel776) [inline doc/UnitTestsAndCI.md:21]: CircleCi works by cloning our git repository into a Ubuntu

File: doc/UnitTestsAndCI.md [hunk 5]
```diff
@@ -18,10 +18,10 @@ We use the standard `unittest` module for testing our python code.  Our unit tes
...
+We use a free continuous integration service called [CircleCi](http://circleci.com), which recompiles and retests our project every time we push a new commit to our [GitHub repo](http://github.com/robojackets/robocup-software).  Circle-ci works by cloning our git repository into a Ubuntu [Docker](https://www.docker.com) Container (similar to a Virtual Machine) each time it receives a notification from GitHub that there are new commits.  It then looks at our circle config file [circle.yml](../circle.yml), which runs other scripts that build and test our code.  Robocup-Software uses [DoCIF](https://github.com/jgkamat/DoCIF) to manage it's docker-based CI.  Triggered by the circle.yml file, DoCIF builds a baseimage for the project (if any setup files changed), runs the tests listed in [config.docif](../config.docif), and sends it's results back to github to be displayed.  On pull requests, you can see the individual logs of any test by clicking 'details' on any check.
```
Comment (spatel776) [inline doc/UnitTestsAndCI.md:21]: its results back to GitHub to be displayed

File: doc/UnitTestsAndCI.md [hunk 6]
```diff
@@ -18,10 +18,10 @@ We use the standard `unittest` module for testing our python code.  Our unit tes
...
 
 There is a build status icon on our GitHub project's main page README that is green when the master branch build succeeds and red when it fails.  This lets us quickly ensure that our project build hasn't been broken.  Our goal is to make sure the master branch always builds.  We push new code to a separate branch, then merge it only once we verify that it passes the tests on circeci.
 
 ## Coverage
 
-For every build, a seperate test is run that checks the percentage of lines in our codebase that are run by tests.  The results of this are sent to [Coveralls](https://coveralls.io/github/RoboJackets/robocup-software) so we can read them easily.  In addition, the percent coverage of the master branch is displayed on the main github repository alongside the status badge.
+For every build, a separate test is run that checks the percentage of lines in our codebase that are run by tests.  The results of this are sent to [Coveralls](https://coveralls.io/github/RoboJackets/robocup-software) so we can read them easily.  In addition, the percent coverage of the master branch is displayed on the main github repository alongside the status badge.
```
Comment (spatel776) [inline doc/UnitTestsAndCI.md:27]: main GitHub repository

**Review by Yudai-8-om — APPROVED** [review]
Reviewed, LGTM!

**Pull request merged**